sklearn/preprocessing/_encoders.py

```diff
@@ -770,8 +782,15 @@ def fit(self, X, y=None):
         """
         # base classes uses _categories to deal with deprecations in
         # OneHoteEncoder: can be removed once deprecations are removed
+        _set_config(assume_finite=True)
+
+        if self.handle_missing == 'separate':
+            self._encode_missing = True
```

Comment:
This can be written as `self._encode_missing = self.handle_missing == 'separate'`


But why is this useful to do, rather than check self.handle_missing directly?

Reply:
I agree-- this doesn't make sense. I removed the `handle_missing` attribute and replaced it with an `encode_missing` attribute, that accepts boolean input rather than string.